proto/task_execution.proto

```diff
@@ -23,6 +23,11 @@ message TaskParameter {
   //data is a directory, if used for an output all the files in the directory
   //will be copied to the storage location
   bool   directory = 5;
+  //if the parameter is an output, should the element be created before executing
```

Comment:
A 'location' is where the file is stored long term, ie an object store url like s3://my-bucket/the_file.txt. The path is there the file is going to be inside the container running on the machine (thus it must be a subpath of one of the declared volumes attached to the container). directory is a boolean hint that these things are referring to a directory, so implementations can do recursive copying and in the case that it is an output directory and the create bool is also true, the system will go ahead an create the directory before running the command.